### PR TITLE
ISPN-9926 Don't run tests for disabled features

### PR DESCRIPTION
**Pull request by diegolovison**
https://issues.redhat.com/browse/ISPN-9926

### CODE REVIEW

**Comment by tristantarrant** [comment]
Lots of failures

**Comment by diegolovison** [comment]
@tristantarrant I have fixed the failures.

File: core/src/test/java/org/infinispan/test/FeaturesListener.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import org.testng.ITestClass;
+import org.testng.ITestContext;
+import org.testng.ITestNGMethod;
+import org.testng.internal.BaseTestMethod;
+
+public class FeaturesListener implements IMethodInterceptor {
+
+   @Override
+   public List<IMethodInstance> intercept(List<IMethodInstance> methods, ITestContext context) {
+      Features features = new Features(this.getClass().getClassLoader());
```
Comment (tristantarrant) [inline core/src/test/java/org/infinispan/test/FeaturesListener.java:20]: Maybe we should use the class loader of the test instance which you are retrieving in the line below.

File: core/src/test/java/org/infinispan/test/FeaturesListener.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      Object instance = methods.get(0).getMethod().getInstance();
+      Annotation[] annotations = instance.getClass().getAnnotations();
```
Comment (tristantarrant) [inline core/src/test/java/org/infinispan/test/FeaturesListener.java:22]: No need to loop through the annotations. Just  use:
```java
AbstractInfinispanTest.FeatureCondition annotation = instance.getClass().getAnnotation(AbstractInfinispanTest.FeatureCondition.class);
if (annotation != null) {
//...
}
```

**Comment by diegolovison** [comment]
@tristantarrant done

**Pull request merged**